coordinate[] waypoints
int64[] costs
